staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer.go

```diff
@@ -134,6 +137,25 @@ func (g *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, svc
 		return nil, err
 	}
 
+	// Services with multiples protocols are not supported by this controller, warn the users and sets
+	// the corresponding Service Status Condition.
+	// https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1435-mixed-protocol-lb
+	if err := checkMixedProtocol(svc.Spec.Ports); err != nil {
```

Comment:
what will this block do on upgrade of a cluster that already had mixed-protocol services which were already handled by previous versions of this controller? is there cleanup to do? will that cleanup be seen as causing an outage?

Reply:
same question applies to UpdateLoadBalancer

Reply:
In cause there is an existing Service Loadbalancer with multiple protocol, that Service will be half broken, it will have a Loadbalancer configured with only the ports from one protocol working.
On an upgrade the new controller logic will fail to update or ensure the loadbalancer, and warn the user, no further operation will be done by the controller on the Service, just set the Status and generte the events

Reply:
For multi-protocol services we haven't ever synced (where the `Status.LoadBalancer.Ingress` list is empty), skipping services we can't satisfy makes sense.

For multi-protocol services we've already synced, it's unclear to me what we should do. Ignoring updates to those services can leave previously exposed ports exposed that would surprise users or cause problems.

Reply:
> For multi-protocol services we haven't ever synced (where the Status.LoadBalancer.Ingress list is empty), skipping services we can't satisfy makes sense.

ok

> For multi-protocol services we've already synced, it's unclear to me what we should do. Ignoring updates to those services can leave previously exposed ports exposed that would surprise users or cause problems.

1. Old cluster, create ServiceA with 2 ports, Port-TCP and Port-UDP
2. Controllers associates a LoadBalancer with ONLY TCP working (It will have GCP resources associated to the Port-UDP port but those resources will be configured with TCP)
3. User doesn't care, he keeps working with that Service without noticing the Port-UDP doesn't work
4. User upgrades
5. New controller tries to Ensure or Update the loadbalancer, it starts informing the user he has a problem with that loadbalancer, only Delete or "Update to remove the Port-UDP" operations are handled by the controller


Your are right, we should not block the update, just warn ... it can leave GCE resources dangling too ... per example, if the user changes the Service Type

Reply:
fixed, and added a test to verify it, it just checks if it doesn't have the condition, and adds it if necessary, but doesn't interrupt the operation